upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -211,26 +237,79 @@ func (e *Elastigroup) Find(c *fi.Context) (*Elastigroup, error) {
 			actual.AssociatePublicIP = fi.Bool(associatePublicIP)
 		}
 
-		if lc.LoadBalancersConfig != nil {
+		// Load balancer.
+		if e.LoadBalancer != nil && lc.LoadBalancersConfig != nil {
 			if lbs := lc.LoadBalancersConfig.LoadBalancers; len(lbs) > 0 {
-				actual.LoadBalancer = &awstasks.LoadBalancer{
-					Name:             lbs[0].Name,
-					LoadBalancerName: lbs[0].Name,
+				t := &awstasks.LoadBalancer{Name: e.LoadBalancer.Name}
+				s, err := t.Find(c)
+				if err != nil {
+					return nil, err
+				}
+				if s != nil && fi.StringValue(s.LoadBalancerName) == fi.StringValue(lbs[0].Name) {
+					actual.LoadBalancer = s
 				}
 			}
 		}
 
+		// IAM instance profile.
 		if lc.IAMInstanceProfile != nil {
-			actual.IAMInstanceProfile = &awstasks.IAMInstanceProfile{Name: lc.IAMInstanceProfile.Name}
+			t := &awstasks.IAMInstanceProfile{Name: lc.IAMInstanceProfile.Name}
+			s, err := t.Find(c)
+			if err != nil {
+				return nil, err
+			}
+			actual.IAMInstanceProfile = s
```

Comment:
As with the Subnets, `Name` is the comparison field, and so I'm not sure why this is necessary